plugins/parsers/form_urlencoded/parser.go

```diff
@@ -9,19 +9,16 @@ import (
 
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/parsers"
 )
 
-var (
-	// ErrNoMetric is returned when no metric is found in input line
-	ErrNoMetric = fmt.Errorf("no metric in line")
-)
+var ErrNoMetric = fmt.Errorf("no metric in line")
 
 // Parser decodes "application/x-www-form-urlencoded" data into metrics
 type Parser struct {
-	MetricName  string
-	DefaultTags map[string]string
-	TagKeys     []string
-	AllowedKeys []string
+	MetricName  string            `toml:"metric_name"`
```

Comment:
I think we never set this directly through the TOML config... Should we make it an unexported field maybe?

Reply:
If I make it an unexported field, we can't set the "MetricName" in the http_listener _v2 tests so not sure if we want to remove that from the tests?

Reply:
Ok, then keep it but replace the TOML tag to
```suggestion
	MetricName  string            `toml:"-"`
```